app/scripts/controllers/network/createJsonRpcClient.js

```diff
@@ -10,22 +10,22 @@ import {
 import { PollingBlockTracker } from 'eth-block-tracker';
 import { SECOND } from '../../../../shared/constants/time';
 
-const inTest = process.env.IN_TEST;
-const blockTrackerOpts = inTest ? { pollingInterval: SECOND } : {};
-const getTestMiddlewares = () => {
-  return inTest ? [createEstimateGasDelayTestMiddleware()] : [];
-};
-
 export default function createJsonRpcClient({ rpcUrl, chainId }) {
+  const blockTrackerOpts = process.env.IN_TEST
+    ? { pollingInterval: SECOND }
+    : {};
   const fetchMiddleware = createFetchMiddleware({ rpcUrl });
   const blockProvider = providerFromMiddleware(fetchMiddleware);
   const blockTracker = new PollingBlockTracker({
     ...blockTrackerOpts,
     provider: blockProvider,
   });
+  const testMiddlewares = process.env.IN_TEST
```

Comment:
For some reason, the test middleware added here is interfering with the running of the tests for `createJsonRpcClient`. I'm not sure whether the middleware is set up correctly or not, but after the promise resolves in `createEstimateGasDelayTestMiddleware`, the middleware after this point is not getting run, so requests through the provider are never completing. 

To get around this in a minimal fashion, we want to disable this middleware only for these network client tests. The problem is that previously we were reading `process.env.IN_TEST` when this file was 
read. This made it impossible to prevent this test middleware from getting added by unsetting that environment variable. By checking for the variable inside of `createJsonRpcClient`, we can unset that variable before initializing the network client in the test helper and then reset it afterward.